politeiad/backendv2/tstorebe/plugins/pi/cmds.go

```diff
@@ -242,25 +242,131 @@ func (p *piPlugin) cmdBillingStatusChanges(token []byte) (string, error) {
 	return string(reply), nil
 }
 
+// cacheDataGet retrieves the data associated with the given token from the
+// memory cache.  If data doesn't exist in cache it returns nil.
+func (p *piPlugin) cacheDataGet(token string) *cacheData {
+	p.cache.Lock()
+	defer p.cache.Unlock()
+
+	return p.cache.data[token]
+}
+
+// cacheVoteStatusSet stores the vote status associated with the given token
+// in cache. If the cache is full and a new entry is being added, the oldest
+// is removed from the cache.
+func (p *piPlugin) cacheVoteStatusSet(token string, status ticketvote.VoteStatusT) {
+	p.cache.Lock()
+	defer p.cache.Unlock()
+
+	// If an entry associated with the proposal already exists in cache
+	// overwrite the vote status.
+	if p.cache.data[token] != nil {
+		p.cache.data[token].voteStatus = &status
+		return
+	}
+
+	// If entry does not exist and cache is fulli, then remove oldest entry
+	if p.cache.entries.Len() == piCacheLimit {
+		// Remove front - oldest entry from entries list.
+		t := p.cache.entries.Remove(p.cache.entries.Front()).(string)
+		// Remove oldest status from map.
+		delete(p.cache.data, t)
+	}
+
+	// Store new status.
+	p.cache.entries.PushBack(token)
+	p.cache.data[token] = &cacheData{
+		voteStatus: &status,
+	}
+}
+
+// cacheProposalStatusSet stores the proposal status associated with
+// the given token in cache. If the cache is full and a new entry is being
+// added, the oldest entry is removed from the cache.
+func (p *piPlugin) cacheProposalStatusSet(token string, status pi.PropStatusT) {
+	p.cache.Lock()
+	defer p.cache.Unlock()
+
+	// If an entry associated with the proposal already exists in cache
+	// overwrite the proposal status.
+	if p.cache.data[token] != nil {
+		p.cache.data[token].proposalStatus = &status
+		return
+	}
+
+	// If entry does not exist and cache is fulli, then remove oldest entry
+	if p.cache.entries.Len() == piCacheLimit {
+		// Remove front - oldest entry from entries list.
+		t := p.cache.entries.Remove(p.cache.entries.Front()).(string)
+		// Remove oldest status from map.
+		delete(p.cache.data, t)
+	}
+
+	// Store new status.
+	p.cache.entries.PushBack(token)
+	p.cache.data[token] = &cacheData{
+		proposalStatus: &status,
+	}
+}
+
 // cmdSummary returns the pi summary of a proposal.
```

Comment:
You're still fetching the record from the backend for these statuses even though it's not required. 

`pi.PropStatusActive, pi.PropStatusCompleted, pi.PropStatusClosed`